### ISSUE TITLE
[question]  Implicit conversions and best matching function

### ISSUE THREAD

**Issue by ZhiqianXia** [KhronosGroup/glslang#2697]
Hi , 
    i find that shader at different version, glslang will give different best matching function. the root reason is  the difference between **findFunction120** with **findFunction400**.   **findFunction400** have the best match alogrithm, and findFunction120 have nothing. 

my question is Can **GL_ARB_gpu_shader5** also behave like the **E_GL_ARB_gpu_shader_fp64**?   there is 
Implicit conversions at case 1, and these rule is similar to ogl spec46. So I think glslang at case 1 should behave like case2. 

case 1:  best choice is choice 1
**#version 150
#extension GL_ARB_gpu_shader5 : require**
   
case 2:  best choice is choice 2
**#version 460**

**shader:**
```

uniform ivec4 u1;
uniform uvec4 u2;
out     vec4  result;
vec4 f(in vec4 a, in vec4 b){ return a * b;}                 // choice 1
vec4 f(in uvec4 a, in uvec4 b){ return vec4(a - b);}   // choice 2

void main()
{
    result = f(u1, u2);
    switch (gl_VertexID)
    {
      case 0: gl_Position = vec4(-1.0, 1.0, 0.0, 1.0); break;
      case 1: gl_Position = vec4( 1.0, 1.0, 0.0, 1.0); break;
      case 2: gl_Position = vec4(-1.0,-1.0, 0.0, 1.0); break;
      case 3: gl_Position = vec4( 1.0,-1.0, 0.0, 1.0); break;
    }
}
```

**glslang code**

```
const TFunction* TParseContext::findFunction(const TSourceLoc& loc, const TFunction& call, bool& builtIn)
{
	....

    else if (version < 120)
        function = findFunctionExact(loc, call, builtIn);
    else if (version < 400)
        function = **extensionTurnedOn**(E_GL_ARB_gpu_shader_fp64) ? findFunction400(loc, call, builtIn) : findFunction120(loc, call, builtIn); 
    else if (explicitTypesEnabled)
        function = findFunctionExplicitTypes(loc, call, builtIn);
    else
        function = findFunction400(loc, call, builtIn);

    return function;
}
```

Ref:
1. https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_gpu_shader5.txt
2. glsl spec ch6.1 ,

**Comment by greg-lunarg** [comment]
I am on vacation this week. I will respond to your issues next week. Best regards.

**Comment by ZhiqianXia** [comment]
@greg-lunarg  kindly review the PR https://github.com/KhronosGroup/glslang/pull/2724. Thanks.

**Comment by greg-lunarg** [comment]
Fixed by #2806
